_posts/2018-09-15-zkSNARK-in-Ethereum.md

```diff
@@ -0,0 +1,61 @@
+
+To mark this milestone, In this post, I'll try to give a short summary of a zero-knowledge proof and try to describe why it can be useful for Ethereum without going into too much detail because it uses pretty advanced elliptic curve cryptography based on elliptic curves over finite fields. But if you're interested in implementation details you can visit [BN](https://github.com/poanetwork/bn) - our implementation of BN128 elliptic curve operations.
+
+### Zero-knowledge proof
+
+Zero-Knowledge Succinct Non-interactive ARgument of Knowledge, or zkSNARKs, is a form of zero-knowledge proof. A zero-knowledge proof is a type of cryptography that allows one party (the prover) to prove to another party (the verifier) that it possesses some information without revealing any details about information itself.
+
+Zero-knowledge proof must have three properties:
+- Completeness. If the statement is true, the verifier will be convinced that the prover is honest.
+- Soundness If the statement is false, nothing can convince the verifier that the statement is true.
+- Zero-knowledge. If the statement is true, the verifier can not learn any additional information from the statement.
+
+#### Example
+
+Let's see the easiest and the most understandable example of a zero-knowledge proof that was inspired by the work of Professor Oded Goldreich.
+
+Imagine that your friend is color-blind (by the way, you can't be color-blind in this example, I'm sorry if you are) and you have two identical green and red balls. Your friend shows you these balls and puts them behind his back. He can swap around the balls behind his back and then he shows them to you again. Your job is to convince him that the balls are of a different color. After many switches, it will be obvious to your friend that the balls are of a different color because your statements will be true every time.
+
+This example satisfies all three zero-knowledge proof properties.
+
+#### Use cases
+
+- Authentication without revealing your identity
+- Proving that you have some private information (for example, private key) without revealing information itself
+- All other NP-class problems
+
+### Zero-knowledge proof in blockchains
+
+We all know that blockchain technology is based on the idea of a decentralized ledger that records transactions and so forth. But the problem in the naive "ledger" approach is that it's transparent. Everybody can see all running transactions, payment history, balances etc. This is where Zero-knowledge proof comes into play.
+
+#### Zcash
+
+Zcash is the first cryptocurrency that based on zero-knowledge proofs. The novelty of Zcash is privacy which is backed by zkSnarks. Its transactions have the option to hide the sender, recipient, and value on the blockchain.
```

Comment:
Zcash is the first cryptocurrency that based on zero-knowledge proofs. Zcash provides privacy options backed by zkSnarks. Transactions can be configured to hide the sender, recipient, and value on the blockchain.